refiner-lib/src/refiner_inner.rs

```diff
@@ -581,20 +595,30 @@ fn build_transaction(
                 }
                 InnerTransactionKind::Deploy | InnerTransactionKind::DeployErc20 => {
                     hash = virtual_receipt_id.0.try_into().unwrap();
-                    tx = tx
-                        .hash(hash)
-                        .from(near_account_to_evm_address(predecessor_id.as_bytes()));
+                    let from_address = near_account_to_evm_address(predecessor_id.as_bytes());
+                    let nonce = storage
+                        .with_engine_access(
+                            near_block.header.height,
+                            transaction_index.try_into().unwrap(),
+                            &[],
+                            |io| aurora_engine::engine::get_nonce(&io, &from_address),
+                        )
+                        .result;
+                    let contract_address = create_legacy_address(&from_address, &nonce);
+
+                    tx = tx.hash(hash).from(from_address);
 
                     tx = tx
                         .to(None)
-                        .nonce(0)
+                        .nonce(nonce.try_into().unwrap())
```

Comment:
Using this conversion because nonce is coming as a U256, but we recently change the tx type to u64. Probably we are fine since the coming nonce should fit in u64 but wanted to make sure. Maybe we should propagate the nonce-to-u64 change to the aurora-engine repo.